ui/src/app/views/account-creation-modals/component.tsx

```diff
@@ -0,0 +1,206 @@
+  }
+
+  updateAndSend(): void {
+    if (this.contactEmailInvalidError()) {
+      return;
+    }
+    this.props.passNewEmail(this.state.contactEmail);
+    const request: UpdateContactEmailRequest = {
+      username: this.props.username,
+      contactEmail: this.state.contactEmail,
+      creationNonce: this.props.creationNonce
+    };
+    const args: FetchArgs = ProfileApiFetchParamCreator().updateContactEmail(request);
+    fetch(this.fullUrl(args.url), args.options).then(() => {
+      this.setState({resendingEmail: false, changingEmail: false});
+    });
+    this.props.closeFunction();
+  }
+
+  send() {
+    const request: ResendWelcomeEmailRequest = {
+      username: this.props.username,
+      creationNonce: this.props.creationNonce
+    };
+    const args: FetchArgs = ProfileApiFetchParamCreator().resendWelcomeEmail(request);
+    fetch(this.fullUrl(args.url), args.options).then(() => {
+      this.setState({resending: false});
+    });
+    this.props.closeFunction();
+  }
+
+  leaveFocusEmail(evt): void {
+    this.setState({emailOffFocus: true, contactEmail: evt.target.value});
+  }
+
+  enterFocusEmail(): void { this.setState({emailOffFocus: false}); }
+
+  updateContactEmail(evt: React.ChangeEvent<HTMLInputElement>): void {
+    this.setState({contactEmail: evt.target.value});
+  }
+
+  contactEmailInvalidError(): boolean {
+    return !(new RegExp(/^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/).test(this.state.contactEmail));
+  }
+
+  showEmailValidationError(): boolean {
+    if (isBlank(this.state.contactEmail) || !this.state.emailOffFocus) {
+      return false;
+    }
+    return this.contactEmailInvalidError();
+  }
+
+  close(): void {
+    this.props.closeFunction();
+  }
+
+  render() {
+    return <React.Fragment>
+        {this.props.update &&
+        <Modal>
+          <ModalTitle>Change contact email</ModalTitle>
+          <ModalBody>
+            <table style={{width: '100%'}}>
+              <tbody>
+              <tr>
+                <td><label>Contact Email:</label></td>
+                <td style={{width: '70%'}}><FieldInput
+                  id={'change-contact-email'}
+                  style={this.showEmailValidationError() ? styles.unsuccessfulInput : {}}
+                  onBlur={(e) => this.leaveFocusEmail(e)}
+                  onFocus={() => this.enterFocusEmail()}
+                >
+                </FieldInput>
+                </td>
+              </tr>
+              <tr>
+                <td></td>
+                <td style={{width: '76%'}}>
+                  {this.showEmailValidationError() &&
+                  <Error>Email is not valid.</Error>
+                  }
+                </td>
+              </tr>
+              </tbody>
+            </table>
+          </ModalBody>
+          <ModalFooter>
+            <button type={'button'} className='btn btn-outline'
+                    onClick={() => this.close()}>Cancel</button>
+            <button id={'change_email'} type={'button'}
```

Comment:
FYI, you don't need braces if the value is a simple string, e.g. `id='change_email' type='button'`